python/TestHarness/TestHarness.py

```diff
@@ -789,6 +803,16 @@ def parseCLArgs(self, argv):
         tmp_str = getattr(self.options, key)
         setattr(self.options, key, value[0])
 
+    # If attempting to test only failed_tests, open the .failed_tests file and create a list object
+    # otherwise, open the failed_tests file object for writing (clobber).
+    self.test_list = []
+    failed_tests_file = os.path.join(os.getcwd(), '.failed_tests')
```

Comment:
Did you test this if you are downstream in a test directory? (e.g. ../../run_tests --re=some_directory). Does it write the failed test to the wrong directory? Instead, use the method we use in `run_tests` to get the location of the script directory, not the cwd.